src/Tracing/Sampling/EventSubscriber/AbstractEventSubscriber.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the worldia/instrumentation-bundle package.
+ * (c) Worldia <developers@worldia.com>
+ */
+
+namespace Instrumentation\Tracing\Sampling\EventSubscriber;
+
+use Instrumentation\Tracing\Sampling\TogglableSampler;
+use Instrumentation\Tracing\Sampling\Voter\VoterInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+
+abstract class AbstractEventSubscriber implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            static::getEventClass() => [['vote', 8092]],
+        ];
+    }
+
+    public function __construct(private TogglableSampler $sampler, private VoterInterface $voter)
+    {
+    }
+
+    public function onRequest(RequestEvent $event): void
```

Comment:
```suggestion
    public function vote(object $event): void
``` ?

Reply:
beware, `CommandEventSubscriber extends AbstractEventSubscriber` and sent events are not compatible with `RequestEvent $event` signature

Reply:
Arg thanks you're right! sorry i read too quickly ;) 
Fixed here https://github.com/worldia/instrumentation-bundle/commit/ccf4bb756b543ef38a980aed0b7740576cef5591
We definitely need to add some tests...